sdk/core/azure-core/src/main/java/com/azure/core/util/polling/PollerFlux.java

```diff
@@ -139,6 +150,68 @@ public PollerFlux(Duration pollInterval,
+     *     this operation will be called with a new {@link PollingContext}.
+     * @param strategy a known strategy for polling a long running operation in Azure
+     * @param pollResponseType the {@link TypeReference} of the response type from a polling call, or BinaryData if raw
+     *                         response body should be kept. This should match the generic parameter {@link U}.
+     * @param resultType the {@link TypeReference} of the final result object to deserialize into, or BinaryData if raw
+     *                   response body should be kept. This should match the generic parameter {@link U}.
+     * @param <T> The type of poll response value.
+     * @param <U> The type of the final result of long running operation.
+     * @return PollerFlux
+     */
+    @SuppressWarnings("unchecked")
+    public static <T, U> PollerFlux<T, U>
+        create(Duration pollInterval,
+               Supplier<Mono<? extends Response<?>>> initialOperation,
+               PollingStrategy<T, U> strategy,
+               TypeReference<T> pollResponseType,
+               TypeReference<U> resultType) {
+        return create(
+            pollInterval,
+            ctx -> initialOperation.get()
+                .flatMap(r -> strategy.canPoll(r).flatMap(canPoll -> {
+                    if (!canPoll) {
+                        return Mono.error(new IllegalStateException(
+                            "Cannot poll with strategy " + strategy.getClass().getSimpleName()));
+                    }
+                    return strategy.onInitialResponse(r, ctx, pollResponseType).flatMap(status -> {
+                        if (r.getValue() == null) {
+                            return Mono.just(new PollResponse<>(status, null));
+                        } else if (TypeUtil.isTypeOrSubTypeOf(r.getValue().getClass(), pollResponseType.getJavaType())) {
+                            return Mono.just(new PollResponse<>(status, (T) r.getValue()));
+                        } else {
+                            Mono<BinaryData> binaryDataMono;
+                            if (r.getValue() instanceof BinaryData) {
+                                binaryDataMono = Mono.just((BinaryData) r.getValue());
+                            } else {
+                                binaryDataMono = BinaryData.fromObjectAsync(r.getValue());
+                            }
+                            if (TypeUtil.isTypeOrSubTypeOf(BinaryData.class, pollResponseType.getJavaType())) {
+                                return binaryDataMono.map(binaryData -> new PollResponse<>(status, (T) binaryData));
+                            } else {
+                                return binaryDataMono.flatMap(binaryData -> binaryData.toObjectAsync(pollResponseType))
+                                    .map(value -> new PollResponse<>(status, value));
+                            }
```

Comment:
It is a bit hard for me to understand this part when `<T>` is not `BinaryData`.

E.g. if the response of the `initalOperation` is `Mono<Response<Flux<ByteBuffer>>>`, and the `<T>` is `PollResult<S>` (which might or might not work well with Jackson), the flow seems to be `Flux<ByteBuffer>` -> `BinaryData` (which is fine) -> `PollResult<S>` (which seems not fine whatever the `<S>`).

Reply:
Another issue, when create `PollResponse` from response of `initalOperation`, I didn't see anywhere it set the `retryAfter`.

I am not sure whether it is not supported here (again, mgmt need to control the interval from activation and the first poll), or it is handled in `poll` part.

Reply:
So maybe the current `onInitialResponse` is a little too simplified (for mgmt use case).

Reply:
Merging this logic into the `onInitialResponse` in the `PollingStrategy` to allow custom polling strategies to customize this.

Reply:
This is moved.